### PR TITLE
Disable index gateway client on read target.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
PR #6461 introduced a fix for the read target that was supposed to
disable the index gateway clients. However, `fallthrough` does not
behave as expected. Thus we are using a nested if statement here.

### CODE REVIEW

**Review by dannykopping — APPROVED** [review]
LGTM, thanks

File: pkg/loki/modules.go [hunk 1]
```diff
@@ -490,8 +490,13 @@ func (t *Loki) initStore() (_ services.Service, err error) {
 			// Use AsyncStore to query both ingesters local store and chunk store for store queries.
 			// Only queriers should use the AsyncStore, it should never be used in ingesters.
 			asyncStore = true
-			fallthrough
-		case t.Cfg.isModuleEnabled(IndexGateway) || t.Cfg.isModuleEnabled(Read):
+
+			if t.Cfg.isModuleEnabled(Read) {
+				// we want to use the actual storage when running the index-gateway, so we remove the Addr from the config
+				t.Cfg.StorageConfig.BoltDBShipperConfig.IndexGatewayClientConfig.Disabled = true
```
Comment (salvacorts) [inline pkg/loki/modules.go:496]: nit: maybe we can extract this to a function. If in the future we make modifications on the `t.Cfg.isModuleEnabled(IndexGateway)` case, they should go inside the if statement as well, but we may skip it unintentionally.

File: pkg/loki/modules_test.go [hunk 2]
```diff
@@ -216,7 +216,50 @@ func TestIndexGatewayRingMode_when_TargetIsRead(t *testing.T) {
 		})
 
 	})
+}
+
+func TestIndexGatewayClientConfig_when_TargetIsQuerier(t *testing.T) {
```
Comment (salvacorts) [inline pkg/loki/modules_test.go:221]: We are testing both the read and the querier target here. So the test name should not be `...TargetIsQuerier`.
Maybe `TestIndexGatewayClientConfig_when_TargetIsQuerierOrRead`?

File: pkg/loki/modules_test.go [hunk 3]
```diff
@@ -216,7 +216,50 @@ func TestIndexGatewayRingMode_when_TargetIsRead(t *testing.T) {
...
+
+		services, err := c.ModuleManager.InitModuleServices(Read)
+		defer func() {
+			for _, service := range services {
+				service.StopAsync()
+			}
+		}()
 
+		require.NoError(t, err)
+		assert.Equal(t, true, c.Cfg.StorageConfig.BoltDBShipperConfig.IndexGatewayClientConfig.Disabled)
```
Comment (salvacorts) [inline pkg/loki/modules_test.go:260]: nit: `assert.True`

File: pkg/loki/modules_test.go [hunk 4]
```diff
@@ -216,7 +216,50 @@ func TestIndexGatewayRingMode_when_TargetIsRead(t *testing.T) {
...
+
+		services, err := c.ModuleManager.InitModuleServices(Querier)
+		defer func() {
+			for _, service := range services {
+				service.StopAsync()
+			}
+		}()
+
+		require.NoError(t, err)
+		assert.Equal(t, false, c.Cfg.StorageConfig.BoltDBShipperConfig.IndexGatewayClientConfig.Disabled)
```
Comment (salvacorts) [inline pkg/loki/modules_test.go:239]: nit: `assert.False`

File: pkg/loki/modules.go (hunk 1 above)
Comment (jeschkies) [reply pkg/loki/modules.go:496]: I think that's a next step :slightly_smiling_face:

**Pull request merged**